File: src/main/java/br/gov/pf/model/entity/State.java
package br.gov.pf.model.entity;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import javax.validation.constraints.NotNull;

@Entity
@Table(name = "states")
public class State extends AbstractEntity {

    @NotNull
    @Column(name = "description")
    private String description;

    @NotNull
    @Column(name = "Abbrev")
    private String abbrev;

    @NotNull
    @ManyToOne
    private Country country;


    public State() {
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getAbbrev() {
        return abbrev;
    }

    public void setAbbrev(String abbrev) {
        this.abbrev = abbrev;
    }

    public Country getCountry() {
        return country;
    }

    public void setCountry(Country country) {
        this.country = country;
    }
}

File: src/main/java/br/gov/pf/model/entity/State.kt
package br.gov.pf.model.entity;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import javax.validation.constraints.NotNull;

@Entity
@Table(name = "states")
class State : AbstractEntity() {

    @NotNull
    @Column(name = "description")
    var description: String = ""

    @NotNull
    @Column(name = "Abbrev")
    var abbrev: String = ""

    @NotNull
    @ManyToOne
    var country: Country = Country()
}
